File: app/src/main/java/com/apanda/base/Module/Gank/presenter/FragmentGirlsPresenter.java
package com.apanda.base.Module.Gank.presenter;


import com.apanda.base.Module.Gank.bean.GilrsBean;
import com.apanda.base.Module.Gank.contract.FragmentGirlsContract;
import com.apanda.base.base.baserx.RxSubscriber;

public class FragmentGirlsPresenter extends FragmentGirlsContract.Presenter {

    public void loadGrilsBean() {
        mRxManage.add(mModel.loadGrilsBean().subscribe(new RxSubscriber<GilrsBean>(mContext, false) {
            @Override
            public void onStart() {
                mView.showLoading("请稍后...");
                super.onStart();
            }

            @Override
            protected void _onNext(GilrsBean _gankBean) {
                mView.returnGrilsBean(_gankBean);
                mView.stopLoading();
            }

            @Override
            protected void _onError(String message) {
                mView.showErrorTip(message);
            }
        }));
    }
}

File: app/src/main/java/com/apanda/base/Module/Gank/presenter/FragmentGirlsPresenter.kt
package com.apanda.base.Module.Gank.presenter


import com.apanda.base.Module.Gank.bean.GilrsBean
import com.apanda.base.Module.Gank.contract.FragmentGirlsContract
import com.apanda.base.base.baserx.RxSubscriber

class FragmentGirlsPresenter : FragmentGirlsContract.Presenter() {

    fun loadGrilsBean() {
        mRxManage.add(mModel.loadGrilsBean().subscribe(object : RxSubscriber<GilrsBean>(mContext!!, false) {
            override fun onStart() {
                mView.showLoading("请稍后...")
                super.onStart()
            }

            override fun _onNext(_gankBean: GilrsBean) {
                mView.returnGrilsBean(_gankBean)
                mView.stopLoading()
            }

            override fun _onError(message: String) {
                mView.showErrorTip(message)
            }
        }))
    }
}